### PR TITLE
Escape JS text in `AppDriver$wait_for_js()`

### LINKED ISSUE #236: `AppDriver$wait_for_js()` does not quote script input
The `eval()` call in the `chromote_wait_for_condition ` function is essentially a NOP instruction as the `condition_js ` script is not quoted: https://github.com/rstudio/shinytest2/blob/61d51d9e96640bfae3eeefa12c4b685099b584c8/R/chromote-methods.R#L105
This leads to e.g. the input script `'test'==='test';` throwing a Javascript error:  `SyntaxError: Unexpected token ';'. Expected ')' to end an argument list.` as the code to be executed looks like this: `eval('test'==='test';)` which is obviously invalid syntax.
To fix this, the `condition_js` needs to be quoted properly.

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #236

TIL `deparse(txt)` is a great to way to escape the text string `txt`. Great trick! Thanks, @jcheng5

--------------------

Could not use `deparse1(txt)` as it was introduced in R 4.0 and we currently support R 3.5, R 3.6.

### CODE REVIEW

File: R/chromote-methods.R [hunk 1]
```diff
@@ -95,14 +95,18 @@ chromote_wait_for_condition <- function(
...
+
   # Must use manual calulation of timeout, as `chromote_session` does not have a
   # way to cancel the `setTimeout` that has already been submitted. (Which will never stop resubmitting)
   script <- paste0(
-# `callback` provided by chromote_execute_script_callback()
 "new Promise((resolve, reject) => {
   let start = Date.now();
   const condition = () => {
-    return eval(", condition_js, ");
+    return eval(", escaped_condition_js, ");
```
Comment (jcheng5) [inline R/chromote-methods.R:109]: Do you want to add a try/catch around the `condition()` call and reject the promise?

Comment (schloerke) [reply R/chromote-methods.R:109]: Good catch! Will add it when the function is called

**Comment by schloerke** [comment]
- [ ] tryCatch

**Comment by jcheng5** [comment]
Are you sure you want to `return false` rather than `reject(err)`?

**Comment by schloerke** [comment]
- [ ] Move try catch and reject on error

**Comment by schloerke** [comment]
Checks good. Merging

**Pull request merged**